istioctl/cmd/istioctl/kubeinject.go

```diff
@@ -412,15 +364,15 @@ func init() {
 	injectCmd.PersistentFlags().BoolVar(&debugMode, "debug", false, "Use debug images and settings for the sidecar")
 
 	deprecatedFlags := []string{"coreDump", "imagePullPolicy", "includeIPRanges", "excludeIPRanges", "hub", "tag",
-		"includeInboundPorts", "excludeInboundPorts", "debug", "verbosity", "sidecarProxyUID", "setVersionString"}
+		"includeInboundPorts", "excludeInboundPorts", "debug", "verbosity", "sidecarProxyUID", "setVersionString",
+		"readinessFailureThreshold", "readinessInitialDelaySeconds", "readinessPeriodSeconds", "rewriteAppProbe",
+		"statusPort"}
```

Comment:
IMO we should remove these flags altogether. "deprecate" implies that they are still usable and support may be removed in the future. However, this PR removes that support. The deprecate prints a warning but don't affect the process error code AFAICT.

Reply:
I thought we were require to deprecate before removing.  If we are allowed to remove without a grace period we should favor removal.

Reply:
Most of these flags were already marked as deprecated for 10 months (see https://github.com/istio/istio/pull/5800). User's had to explicitly define an environment variable and set hub/tag to use them, otherwise they would get an error. Now that this PR removes the underlying functionality we should also remove the flags.

Reply:
cc @mandarjog

Reply:
The normal procedure is to deprecate first, and after at least one release or more, remove the flag and its support.

Reply:
So should i remove these flags in this PR? Or deprecate these flags at first and hold the PR until next release?

Reply:
I followed @ayj PR link and I agreed it's okay to remove them at all and won't cause not major impact.